homeassistant/components/network/__init__.py

```diff
@@ -45,6 +46,35 @@ async def async_get_source_ip(hass: HomeAssistant, target_ip: str) -> str:
     return source_ip if source_ip in all_ipv4s else all_ipv4s[0]
 
 
+@bind_hass
+async def async_get_enabled_source_ips(
+    hass: HomeAssistant,
+) -> list[IPv4Address | IPv6Address]:
+    """Build the list of enabled source ips."""
+    adapters = await async_get_adapters(hass)
+    sources: list[IPv4Address | IPv6Address] = []
+    for adapter in adapters:
+        if not adapter["enabled"]:
+            continue
+        if adapter["ipv4"]:
+            sources.extend(IPv4Address(ipv4["address"]) for ipv4 in adapter["ipv4"])
+        if adapter["ipv6"]:
+            # With python 3.9 add scope_ids can be
+            # added by enumerating adapter["ipv6"]s
+            # IPv6Address(f"::%{ipv6['scope_id']}")
+            sources.extend(IPv6Address(ipv6["address"]) for ipv6 in adapter["ipv6"])
+
+    return sources
+
+
+@callback
+def async_use_default_interface(adapters: list[Adapter]) -> bool:
```

Comment:
I wonder if we can improve the naming a bit. Maybe something like `async_only_default_interface_enabled` ?

Reply:
Works for me. 👍